webserver/observe/metrics/src/main/java/io/helidon/webserver/observe/metrics/PostRequestMetricsSupportImpl.java

```diff
@@ -24,10 +24,25 @@
 
 class PostRequestMetricsSupportImpl implements PostRequestMetricsSupport {
 
+    private static final System.Logger LOGGER = System.getLogger(PostRequestMetricsSupportImpl.class.getName());
+
+    private static boolean loggedMessageAboutMissingKpiDataStructure = false;
```

Comment:
Naïve comment: should this be `volatile` (and possibly atomic via `VarHandle`)? Don't know if multiple threads can get here but if they can then the message might be output twice if there's a race

Reply:
No, not naïve and probably a valid point. 

But I hope this code will vanish when we (I guess that means when I) revisit how we deal with KPI metrics on multiple sockets. If the server output contains a small number >1 copies of the warning due to a heavy initial rush of incoming requests then I'm very OK with that.